### ISSUE TITLE
Network profile for mapping open ports to localhost

### ISSUE THREAD

**Issue by techtonik** [lxc/lxd#3532]
* LXD version: 2.15

# Issue description

[nic device](https://github.com/lxc/lxd/blob/fa06a11035d5f61b44a9b2c60d25f334a85d8dfa/doc/containers.md#type-nic) configuration doesn't expose the most frequent option in development setup to map open ports inside VM to local ports on the host.

It would be very nice to have an option to provide such port mapping for container during run-time. Is that technically feasible with network profiles?

**Comment by stgraber** [comment]
That's what #2504 would allow, among other things.

**Comment by techtonik** [comment]
I would keep this story open, because it is about next step - shipping profile once #2504 is ready.
